File: jdk_17_maven/cs/rest/tiltaksgjennomforing-api/src/main/java/no/nav/tag/tiltaksgjennomforing/featuretoggles/FeatureToggleController.java
package no.nav.tag.tiltaksgjennomforing.featuretoggles;

import java.util.List;
import java.util.Map;
import io.getunleash.Variant;
import no.nav.security.token.support.core.api.Unprotected;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@Unprotected
@RequestMapping("/feature")
public class FeatureToggleController {
    private final FeatureToggleService featureToggleService;

    @Autowired
    public FeatureToggleController(FeatureToggleService featureToggleService) {
        this.featureToggleService = featureToggleService;
    }

    @GetMapping
    public Map<String, Boolean> feature(@RequestParam("feature") List<String> features) {
        return featureToggleService.hentFeatureToggles(features);
    }

    @GetMapping("/variant")
    public Map<String, Variant> variant(@RequestParam("feature") List<String> features) {
        return  featureToggleService.hentVarianter(features);
    }

}

File: jdk_17_maven/cs/rest/familie-ba-sak/src/main/kotlin/no/nav/familie/ba/sak/config/featureToggle/FeatureToggleController.kt
package no.nav.familie.ba.sak.config.featureToggle

import no.nav.familie.ba.sak.common.RessursUtils
import no.nav.familie.ba.sak.config.FeatureToggleService
import no.nav.familie.kontrakter.felles.Ressurs
import no.nav.security.token.support.core.api.ProtectedWithClaims
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/feature")
@ProtectedWithClaims(issuer = "azuread")
@Validated
class FeatureToggleController(
    private val featureToggleService: FeatureToggleService,
) {

    @PostMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
    fun hentToggles(@RequestBody toggles: List<String>): ResponseEntity<Ressurs<Map<String, Boolean>>> {
        return RessursUtils.ok(
            toggles.fold(mutableMapOf()) { acc, toggleId ->
                acc[toggleId] = featureToggleService.isEnabled(toggleId)
                acc
            },
        )
    }
}
